SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc

```diff
@@ -699,16 +703,24 @@ void SiPixelDigitizerAlgorithm::digitize(const PixelGeomDetUnit* pixdet,
     if(pixdet->type().isTrackerPixel() && pixdet->type().isBarrel()){ // Barrel modules
       int lay = tTopo->layer(detID);
       if(addThresholdSmearing) {
-	if((pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)  && lay==1) {
-	  thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix_L1, theThresholdSmearing_BPix_L1); // gaussian smearing
-	} else {
-	  thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix , theThresholdSmearing_BPix); // gaussian smearing
+	if(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB) {
+	  if (lay==1) {
+	    thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix_L1, theThresholdSmearing_BPix_L1); // gaussian smearing
+	  } else if (lay==2) {
+	    thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix_L2, theThresholdSmearing_BPix_L2); // gaussian smearing
+	  } else {
+	    thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix , theThresholdSmearing_BPix); // gaussian smearing
+	  }
 	}
       } else {
-	if((pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)  && lay==1) {
-	  thePixelThresholdInE = theThresholdInE_BPix_L1;
-	} else {
-	  thePixelThresholdInE = theThresholdInE_BPix; // no smearing
+	if(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB) {
```

Comment:
Just a curiosity appeared while looking at the code changes...

With the previous code, if
!(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || 
            pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
thePixelThresholdInE would have been set = theThresholdInE_BPix

With the current code if
!(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || 
            pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
thePixelThresholdInE remains 0.

My question: can 
(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel ||
 pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
be false?
  - if the answer is "yes", please check the logic if actually corresponds to what you want;
  - if the answer is "no", why do you bother with that check?

Reply:
> can
> (pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel ||
> pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
> be false?

I'd say yes: given that the code is within
```cpp
 if(pixdet->type().isTrackerPixel() && pixdet->type().isBarrel()){ // Barrel modules
```
the `subDetector()` can be `P2PXB`.

Reply:
Indeed... Although I did not see any change in the Phase2 workflows outputs in jenkins, that's why I'm asking if there is anything that prevents Phase2 code to access this method: I did not check that deep in CMSSW to verify it,

Reply:
Let me add some phase2 experts @boudoul @atricomi @delaere

Reply:
hi @perrotta and @makortel , for phase2 a difference digitizer is called thanks to the phase2 Era : https://github.com/cms-sw/cmssw/blob/master/SimGeneral/MixingModule/python/pixelDigitizer_cfi.py#L14